drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -321,13 +321,58 @@ function DragAndDropTemplates(configuration) {
         )
     };
 
-    var mainTemplate = function(ctx) {
-        var problemTitle = ctx.show_title ? h('h2.problem-title', {innerHTML: ctx.title_html}) : null;
-        var problemHeader = ctx.show_problem_header ? h('h3.title1', gettext('Problem')) : null;
-        var popupSelector = 'div.popup';
-        if (ctx.popup_html && !ctx.last_action_correct) {
+    var itemFeedbackPopupTemplate = function(ctx) {
+        var popupSelector = 'div.popup.item-feedback-popup';
+        var msgs = ctx.feedback_messages || [];
+        var have_messages = msgs.length > 0;
+        var popup_content;
+
+        var close_button_describedby_id = "close-popup-"+configuration.url_name;
+
+        if (msgs.length > 0 && !ctx.last_action_correct) {
             popupSelector += '.popup-incorrect';
         }
+
+        if (ctx.mode == DragAndDropBlock.ASSESSMENT_MODE) {
+            var content_items = [
+                (!ctx.last_action_correct) ? h("p", {}, gettext("Some of your answers were not correct")) : null,
+                h("p", {}, gettext("Hints:")),
+                h("ul", {}, msgs.map(function(message) {
+                    return h("li", {innerHTML: message.message});
+                }))
+            ];
+            popup_content = h("div.popup-content", {}, have_messages ? content_items : []);
+        } else {
+            popup_content = h("div.popup-content", {}, msgs.map(function(message) {
+                return h("p", {innerHTML: message.message});
+            }))
+        }
+
+        return h(
+            popupSelector,
+            {
+                style: {display: have_messages ? 'block' : 'none'},
+                attributes: {"tabindex": "-1"}
+            },
+            [
+                h(
```

Comment:
I am not accustomed to seeing a button done this way. I suggest you follow the pattern here:

http://ux.edx.org/components/buttons/
and
http://ux.edx.org/design_elements/icons/

So there would be a button containing 2 spans-- one that has the icon (always aria-hidden), and one with the sr text. Similar to this--

![image](https://cloud.githubusercontent.com/assets/484484/17782764/17e5780e-6543-11e6-809f-c746418d1392.png)